packages/react-router/modules/__tests__/Route-test.js

```diff
@@ -34,6 +34,23 @@ describe('A <Route>', () => {
 
     expect(node.innerHTML).toNotContain(TEXT)
   })
+
+  it('supports preact by nulling out children prop when empty array is passed', () => {
+    const TEXT = 'Mrs. Kato'
+    const node = document.createElement('div')
+
+    ReactDOM.render((
+      <MemoryRouter initialEntries={[ '/' ]}>
+        <Route path="/" render={() => (
+          <h1>{TEXT}</h1>
+        )}>
+          {[]}
```

Comment:
@mjackson yes. `props.children` is always a flat Array with in Preact, with empties removed and adjacent Strings merged.  The empty children Array is always the same shared reference to an empty Array so that `shallowEqual` checks within `shouldComponentUpdate` will correctly ignore repeated empty children (for leaf nodes).